rpc/server.go

```diff
@@ -0,0 +1,91 @@
+package rpc
+
+import (
+	"github.com/hashicorp/yamux"
+	log "github.com/sirupsen/logrus"
+	"net"
+	"net/rpc"
+	"net/rpc/jsonrpc"
+	"sync"
+
+	"github.com/thunderdb/ThunderDB/proto"
+)
+
+type ServiceMap map[string]interface{}
+
+type Server struct {
+	node       *proto.Node
+	sessions   sync.Map // map[id]*Session
+	rpcServer  *rpc.Server
+	stopCh     chan interface{}
+	serviceMap ServiceMap
+}
+
+func NewServer() *Server {
+	return &Server{
+		rpcServer:  rpc.NewServer(),
+		stopCh:     make(chan interface{}),
+		serviceMap: make(ServiceMap),
+	}
+}
+
+func NewServerWithService(serviceMap ServiceMap) (server *Server, err error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewServerWithService should have comment or be unexported